code/game/objects/items/candle.dm

```diff
@@ -69,6 +70,32 @@
 		processing_objects.Add(src)
 		update_icon()
 
+/obj/item/candle/proc/flicker(var/amount = rand(5, 15))
+	if(flickering)
+		return
+	flickering = 1
+	spawn(0)
+		if(lit)
+			for(var/i = 0; i < amount; i++)
+				lit = 0
+				update_icon()
+				sleep(rand(3,9))
+				lit = 1
```

Comment:
you could get a better flicker effect by instead adjusting the light power or radius (or both)

Reply:
Fine, I'll see what I can do.

Reply:
done